src/Microsoft.DotNet.VersionTools/Automation/LocalVersionsRepoUpdater.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
You have a better sense for this than I, but I'll ask anyway.  Did you consider having an abstract base class that housed logic like `CreatePackageListFile` and the gathering of packages with two subclasses, GitHubVersionsRepoUpdater and LocalVersionsRepoUpdater which were just concerned about actually pushing the txt files around?  Is the overhead not worth it for the sharing of the linq clauses and argument checking?